### ISSUE TITLE
head_d_cls_pyramid_pooling raise an error

### ISSUE THREAD

**Issue by visonpon** [lukasliebel/MultiDepth#3]
Hi @lukasliebel 
Thanks for your great work, I got an error at pyramid pooling layer no matter how to change the image size , it always triggers a value error: `Expected more than 1 value per channel when training, got input size torch.size([1,512,1,1])`

Hope you can give some advice , thanks~

**Comment by lukasliebel** [comment]
Hi there,

thanks for your interest in our work!

So according to the error message you posted, you are using a batch size of 1. This error is most likely caused by a BatchNorm layer which cannot operate on a single sample (see this discussion: https://discuss.pytorch.org/t/error-expected-more-than-1-value-per-channel-when-training/26274). Please try increasing the batch size to at least 2 to get rid of the error but keep in mind that BatchNorm only really makes sense for larger batches (cf. its formal definition: https://arxiv.org/pdf/1502.03167.pdf and implementation: https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm2d.html#torch.nn.BatchNorm2d)

Hope this helps.
Cheers,
Lukas

**Comment by visonpon** [comment]
thanks for your quick response, after change the Batch size, it's normal to train ~

**Comment by lukasliebel** [comment]
Great. Glad to hear that! I'll close this issue for now. Feel free to let me know if you have any other questions. If you can make use of this concept or implementation in your work, I'd be happy to hear about it as well :)
